src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -957,30 +969,55 @@ void readAnalyze(TrackPointer track, double sampleRate, int timingOffset, bool i
-        pLoadCue->setLabel(firstCue.comment);
-
-        // Add remaining memory cues as hot cues (after actual found hotcues) as Mixxx can only have 1 cue
-        for (int memoryCueIndex = 1; memoryCueIndex < memoryCues.size(); memoryCueIndex++) {
-            memory_cue_t memoryCue = memoryCues[memoryCueIndex];
-            setHotCue(track, memoryCue.position, lastHotCueIndex + memoryCueIndex, memoryCue.comment, memoryCue.color);
+        bool mainCueFound = false;
+        bool firstLoopFound = false;
+
+        // Add memory cues and loops
+        for (int memoryCueLoopIndex = 0; memoryCueLoopIndex < memoryCuesLoops.size(); memoryCueLoopIndex++) {
+            memory_cue_loop_t memoryCueLoop = memoryCuesLoops[memoryCueLoopIndex];
+
+            if (memoryCueLoop.endPosition == -1) {
+                if (mainCueFound) {
+                    // Add remaining memory cues as hot cues (after actual found hotcues) as Mixxx can only have 1 cue
+                    setHotCue(track, memoryCueLoop.startPosition, -1, ++lastHotCueIndex, memoryCueLoop.comment, mixxx::RgbColor::nullopt(), memoryCueLoop.colorID);
+                } else {
+                    // Set first chronological memory cue as Mixxx MainCue
+                    track->setCuePoint(CuePosition(memoryCueLoop.startPosition));
+                    CuePointer pMainCue = track->findCueByType(mixxx::CueType::MainCue);
+                    pMainCue->setLabel(memoryCueLoop.comment);
+                    if (memoryCueLoop.colorID != -1) {
+                        pMainCue->setColor(colorFromID(memoryCueLoop.colorID));
+                    }
+                    mainCueFound = true;
+                }
+            } else {
+                if (firstLoopFound) {
+                    // Add remaining loops as hot cues (after actual found hotcues) as Mixxx can only have 1 loop
+                    // Mixxx v2.4 will feature multiple loops, so these saved here will be usable 
+                    setHotCue(track, memoryCueLoop.startPosition, memoryCueLoop.endPosition, ++lastHotCueIndex, memoryCueLoop.comment, mixxx::RgbColor::nullopt(), memoryCueLoop.colorID);
+                } else {
+                    // Set first chronological loop as Mixxx loop
+                    CuePointer pLoop = track->findCueByType(mixxx::CueType::Loop);
+                    if (!pLoop) {
+                        pLoop = CuePointer(track->createAndAddCue());
+                    }
+                    pLoop->setStartPosition(memoryCueLoop.startPosition);
+                    pLoop->setEndPosition(memoryCueLoop.endPosition);
+                    pLoop->setLabel(memoryCueLoop.comment);
+                    if (memoryCueLoop.colorID != -1) {
+                        pLoop->setColor(colorFromID(memoryCueLoop.colorID));
+                    }
+                    pLoop->setType(mixxx::CueType::Loop);
+                    firstLoopFound = true;
+                }
```

Comment:
Don't add a special case here for the first loop, just imported as a cue with CueType::Loop and the appropriate hotcue number. Mixxx 2.3 will load the first one as the single loop Mixxx supports.